crates/pop-cli/src/common/contracts.rs

```diff
@@ -138,4 +151,40 @@ mod tests {
 		assert!(has_contract_been_built(Some(&path.join(name))));
 		Ok(())
 	}
+
+	#[tokio::test]
+	async fn check_contracts_node_and_prompt_works() -> anyhow::Result<()> {
+		//let cache_path: PathBuf = crate::cache()?;
+
+		let cache_path = tempfile::tempdir().expect("Could create temp dir");
```

Comment:
I'm afraid the cache is going to be a problem in CI. `check_contracts_node_and_prompt` at line 165 returns the cache directory, so can't really just go with a temp.

Reply:
Applied refactor as in https://github.com/r0gue-io/pop-cli/pull/315/files#diff-5110116ab6696521a83af6b4954dacf695aeb5e3f2aa5cd7c2c2755b1a63ec64R15